internal/server/server.go

```diff
@@ -127,56 +128,31 @@ func Serve(
 		return err
 	}
 
-	baseMux := http.NewServeMux()
-	baseMux.Handle("/", gwmux)
+	httpMux := http.NewServeMux()
+	httpMux.Handle("/", gwmux)
 
-	httpServer := &http.Server{
-		Handler:      grpcHandlerFunc(grpcServer, baseMux),
-		Addr:         address,
-		ReadTimeout:  5 * time.Second,
-		WriteTimeout: 10 * time.Second,
-		IdleTimeout:  120 * time.Second,
+	logger.Info("server started")
+	if err := mux.Serve(
+		ctx,
+		mux.WithHTTPTarget(config.addr(), &http.Server{
+			Handler:      grpcHandlerFunc(grpcServer, httpMux),
+			ReadTimeout:  5 * time.Second,
+			WriteTimeout: 10 * time.Second,
+			IdleTimeout:  120 * time.Second,
+		}),
+		mux.WithGRPCTarget(config.grpcAddr(), grpcServer),
+		mux.WithGracePeriod(5*time.Second),
+	); err != ctx.Err() {
+		logger.Error("mux serve error", "err", err)
 	}
 
-	idleConnsClosed := make(chan struct{})
-	interrupt := make(chan os.Signal, 1)
-	go func() {
-		signal.Notify(interrupt, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
-		<-interrupt
-
-		if httpServer != nil {
-			// We received an interrupt signal, shut down.
-			logger.Warn("stopping http server...")
-			if err := httpServer.Shutdown(context.Background()); err != nil {
-				logger.Error("HTTP server Shutdown", "err", err)
-			}
-		}
-
-		if grpcServer != nil {
-			logger.Warn("stopping grpc server...")
-			grpcServer.GracefulStop()
+	if pgClient != nil {
```

Comment:
Might be better to keep this as a deferred function so that it runs even if there is a panic.